### PR TITLE
SOF-2212 IngestUpdateEvents creates entities if missing

### PR DESCRIPTION
**Pull request by LindvedKrvang**
There is an extremely low chance that an IngestCreateEvent is missed. If it is missed, then following IngestUpdateEvents won't have an entity to update and we run into a "stalemate" untill a new IngestCreateEvent is received (either delete Rundown and reingest or float/unfloat the Segment).

This PR introduces that if we receive an IngestUpdateEvent for a Part or Segment that doesn't exist, we then create the Part or Segment instead.

### CODE REVIEW

File: src/business-logic/services/ingest-data-changed-service.ts [hunk 1]
```diff
@@ -29,6 +29,20 @@ import { ActionEventEmitter } from './interfaces/action-event-emitter'
 
 const BULK_EXECUTION_TIMESPAN_IN_MS: number = 500
 
+const enum IngestEventPriority {
```
Comment (KvelaGorrrrnio) [inline src/business-logic/services/ingest-data-changed-service.ts]: Is the const intentional? :)

File: src/business-logic/services/ingest-data-changed-service.ts [hunk 2]
```diff
@@ -433,6 +453,14 @@ export class IngestDataChangedService implements DataChangeService {
     await this.persistRundown(rundown)
   }
 
+  private async fetchSegmentIfExist(segmentId: string): Promise<Segment | undefined> {
+    try {
+      return this.segmentRepository.getSegment(segmentId)
+    } catch (error) {
+      return Promise.resolve(undefined)
+    }
```
Comment (KvelaGorrrrnio) [inline src/business-logic/services/ingest-data-changed-service.ts]: Suggestion: Use the promise syntax to make it a bit cleaner:
```typescript
    return this.segmentRepository.getSegment(segmentId).catch(() => undefined)
```

File: src/business-logic/services/ingest-data-changed-service.ts [hunk 3]
```diff
@@ -433,6 +453,14 @@ export class IngestDataChangedService implements DataChangeService {
     await this.persistRundown(rundown)
   }
 
+  private async fetchSegmentIfExist(segmentId: string): Promise<Segment | undefined> {
```
Comment (KvelaGorrrrnio) [inline src/business-logic/services/ingest-data-changed-service.ts]: Does fetch have a different meaning than get? If not, I suggest that we use `getSegmentIfExists`.

**Review by KvelaGorrrrnio — APPROVED** [review]
Looks good. A few suggestions :)

File: src/business-logic/services/ingest-data-changed-service.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/business-logic/services/ingest-data-changed-service.ts]: Force of habit. It has been removed

File: src/business-logic/services/ingest-data-changed-service.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/business-logic/services/ingest-data-changed-service.ts]: I like it. Updated to use it and removed the other methods.

File: src/business-logic/services/ingest-data-changed-service.ts (hunk 3 above)
Comment (LindvedKrvang) [reply src/business-logic/services/ingest-data-changed-service.ts]: Not really. But the methods has been removed in favour of the promise syntax

**Review by KvelaGorrrrnio — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**